### PR TITLE
Update to Go 1.17

### LINKED ISSUE #885: Research/Upgrade to Go 1.17
Go 1.17 was released and we may see some performance improvements if we upgrade:

https://golang.org/doc/go1.17

### PR DESCRIPTION
**Pull request by zalegrala**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #885

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@
 * [CHANGE] Change default tempo port from 3100 to 3200 [#770](https://github.com/grafana/tempo/pull/809) (@MurzNN)
 * [CHANGE] Jsonnet: use dedicated configmaps for distributors and ingesters [#775](https://github.com/grafana/tempo/pull/775) (@kvrhdn)
 * [CHANGE] Docker images are now prefixed by their branch name [#828](https://github.com/grafana/tempo/pull/828) (@jvrplmlmn)
+* [CHANGE] Update to Go 1.16 [#???](https://github.com/grafana/tempo/pull/???)
```
Comment (mdisibio) [inline CHANGELOG.md:48]:
```suggestion
* [CHANGE] Update to Go 1.17 [#???](https://github.com/grafana/tempo/pull/???)
```

File: .drone/drone.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ steps:
 
 # binaries
 - name: build-tempo-binaries
-  image: golang:1.16.0-alpine
+  image: golang:1.17.0-alpine
```
Comment (mdisibio) [inline .drone/drone.yml:20]: 1.17.1 is already released. Should we just track latest 1.17?
```suggestion
  image: golang:1.17-alpine
```

File: .drone/drone.yml [hunk 3]
```diff
@@ -87,7 +87,7 @@ steps:
...
```
Comment (mdisibio) [inline .drone/drone.yml:90]: 1.17.1 is already released. Should we just track latest 1.17?
```suggestion
  image: golang:1.17.0-alpine
```

File: go.mod [hunk 4]
```diff
@@ -74,6 +74,203 @@ require (
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
 )
 
+require (
```
Comment (mdisibio) [inline go.mod:77]: I'm not familiar with the go mod changes in 1.17. What is this second require section for, is it different than the first?

Comment (zalegrala) [reply go.mod:77]: Paragraph 4 here has what you're after.  https://golang.org/doc/go1.17#go-command  New section to track indirect dependencies.

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**